src/proto/h1/date.rs

```diff
@@ -52,18 +52,18 @@ impl CachedDate {
     }
 
     fn check(&mut self) {
-        let now = time::get_time();
+        let now = OffsetDateTime::now();
         if now > self.next_update {
             self.update(now);
         }
     }
 
-    fn update(&mut self, now: time::Timespec) {
+    fn update(&mut self, now: OffsetDateTime) {
         self.pos = 0;
-        let _ = write!(self, "{}", time::at_utc(now).rfc822());
+        let _ = write!(self, "{}", now.format("%a, %d %b %Y %H:%M:%S GMT"));
```

Comment:
This needs to be explicitly set to UTC. Without this, the output value may not be semantically correct.

```suggestion
        let _ = write!(self, "{}", now.to_offset(UtcOffset::UTC).format("%a, %d %b %Y %H:%M:%S GMT"));
```

Reply:
The `OffsetDateTime::now()` returns the current date and time in UTC.
I don't think it's necessary to set to UTC explicitly.

Reply:
It's zero-cost to convert to UTC if it already is, so I think it would be smart to add it anyways. Otherwise you're in premature optimization territory, as you may as well only be storing the `PrimitiveDateTime` if you're already assuming the time zone.

Reply:
Well, setting the offset to UTC explicitly is not bad at least. I will add it :)